pkg/utils/managedresources/managedresources.go

```diff
@@ -43,39 +109,55 @@ func CreateManagedResourceFromUnstructured(ctx context.Context, client client.Cl
 		data = append(data, []byte("\n---\n")...)
 		data = append(data, bytes...)
 	}
-	return CreateManagedResource(ctx, client, namespace, name, class, name, data, keepObjects, injectedLabels, false)
+	return Create(ctx, client, namespace, name, secretNameWithPrefix, class, map[string][]byte{name: data}, &keepObjects, injectedLabels, pointer.BoolPtr(false))
 }
 
-// CreateManagedResource creates a managed resource and its secret with the given name, class, key, and data in the given namespace.
-func CreateManagedResource(ctx context.Context, client client.Client, namespace, name, class, key string, data []byte, keepObjects bool, injectedLabels map[string]string, forceOverwriteAnnotations bool) error {
-	if key == "" {
-		key = name
-	}
+// Create creates a managed resource and its secret with the given name, class, key, and data in the given namespace.
+func Create(ctx context.Context, client client.Client, namespace, name string, secretNameWithPrefix bool, class string, data map[string][]byte, keepObjects *bool, injectedLabels map[string]string, forceOverwriteAnnotations *bool) error {
+	var (
+		secretName, secret = NewSecret(client, namespace, name, data, secretNameWithPrefix)
+		managedResource    = New(client, namespace, name, class, keepObjects, nil, injectedLabels, forceOverwriteAnnotations).WithSecretRef(secretName)
+	)
 
-	// Create or update secret containing the rendered rbac manifests
-	if err := manager.NewSecret(client).
-		WithNamespacedName(namespace, name).
-		WithKeyValues(map[string][]byte{key: data}).
-		Reconcile(ctx); err != nil {
-		return errors.Wrapf(err, "could not create or update secret '%s/%s' of managed resources", namespace, name)
+	return deployManagedResource(ctx, secret, managedResource)
+}
+
+// CreateForSeed deploys a ManagedResource CR for the seed's gardener-resource-manager.
+func CreateForSeed(ctx context.Context, client client.Client, namespace, name string, keepObjects bool, data map[string][]byte) error {
+	var (
+		secretName, secret = NewSecret(client, namespace, name, data, true)
+		managedResource    = NewForSeed(client, namespace, name, keepObjects).WithSecretRef(secretName)
```

Comment:
Why don't we call New() directly here. I feel `NewForSeed` and `NewForShoot` does not help readability.

Reply:
We could replace `NewForSeed` with a direct `New` call and eliminate `NewForSeed` entirely (it's only used here). However, there is also `NewForShoot` which is called also at some other place, hence, it cannot be eliminated. In order to not duplicate things, I would rather keep it as is, i.e., calling `NewForShoot` here. For similarity reasons, I would also keep `NewForSeed` (even if it's not strictly needed today, but it also doesn't harm//not sure why this should impact readability?).

Reply:
Ok, let's leave it then.